lib/bounce_email.rb

```diff
@@ -219,11 +219,13 @@ def check_if_bounce(mail)
     end
 
     def get_original_mail(mail) #worked alright for me, for sure this has to be extended
+      original =
       if mail.multipart?
         ::Mail.new(mail.parts.last)
       elsif i = index_of_original_message_delimiter(mail)
         ::Mail.new(extract_original_message_after_delimiter(mail, i))
       end
+      extract_and_assign_fields_from_original_mail(original) if original
```

Comment:
Somehow I don't like that implicit return of the modified email object here. Not sure if I have a better suggestion, though.